[Guided onboarding] Add guided onboarding to the home page

yuliacech opened the pull request:
## Summary

Fixes https://github.com/elastic/kibana/issues/138654
Fixes https://github.com/elastic/kibana/issues/138550

This PR adds guided onboarding changes to the home page for Cloud deployments:
- The welcome screen is replaced with the guided onboarding landing page
- A guided onboarding link is added to the integrations section

### How to test
1. Add `xpack.cloud.id: 'testID'` to your `kibana.dev.yml` file to imitate a Cloud deployment
2. Remove `home:welcome:show` value from your `localStorage`
3. Open Kibana and check that you're correctly redirected to the guided onboarding landing page
4. Navigate to the home page and check the new guided onboarding link

### Screenshots

#### Welcome screen for self managed 

<img width="652" alt="Screenshot 2022-10-25 at 12 26 18" src="https://user-images.githubusercontent.com/6585477/197750988-a2cfcc7a-b47b-4ef5-aa49-6ae88916b481.png">


#### Welcome screen for Cloud deployments (guided onboarding landing page)

<img width="1245" alt="Screenshot 2022-10-24 at 15 02 46" src="https://user-images.githubusercontent.com/6585477/197531782-79776abb-200c-4acb-9bd5-bf35a78f2d92.png">


#### Home page with guided onboarding link (large screens)

<img width="1634" alt="Screenshot 2022-10-25 at 12 24 10" src="https://user-images.githubusercontent.com/6585477/197751101-6c103deb-14db-4b71-801e-e8bc099575e1.png">



#### Home page with guided onboarding link (small screens)

<img width="666" alt="Screenshot 2022-10-25 at 12 24 33" src="https://user-images.githubusercontent.com/6585477/197751285-edbe30c7-a97d-47a5-9618-1123f6673d77.png">




### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

elasticmachine commented:
Pinging @elastic/platform-onboarding (Team:Journey/Onboarding)

alisonelizabeth commented on src/plugins/home/public/application/components/add_data/add_data.tsx:
```diff
@@ -66,19 +67,38 @@ export const AddData: FC<Props> = ({ addBasePath, application, isDarkMode }) =>
+                  >
+                    <FormattedMessage
+                      id="home.addData.guidedOnboardingLinkLabel"
+                      defaultMessage="Launch Guided setup"
```
I'll defer to @kellyemurphy, but I think it makes sense for the text to match the button label in the header, so it would be something like: "Launch Setup guide"

yuliacech replied:
Yes, this might have been an older wording, let's see what @kellyemurphy says

kellyemurphy replied:
@alisonelizabeth is right on the money here.

yuliacech replied:
Thank you, @kellyemurphy! I updated the button label and used sentence case for it "Launch setup guide". Is that correct or should I change to "Launch Setup guide"?

alisonelizabeth commented on src/plugins/home/public/application/components/add_data/add_data.tsx:
```diff
@@ -66,19 +67,38 @@ export const AddData: FC<Props> = ({ addBasePath, application, isDarkMode }) =>
                   <EuiButton
                     data-test-subj="homeAddData"
-                    fill
+                    fill={!isCloudEnabled}
```
[nit] Might be helpful to add a comment here to explain why the `fill` value changes on cloud.

alisonelizabeth commented on src/plugins/home/public/application/components/home.test.tsx:
```diff
@@ -230,6 +231,14 @@ describe('home', () => {
+
+      const component = await renderHome();
+
+      expect(component.find(Welcome).exists()).toBe(false);
```
Does it make sense to add an another `expect` statement to verify the getting started page renders?

yuliacech replied:
yes, I was not testing for the correct case here, good catch 👍

alisonelizabeth reviewed (commented):
Thanks for working on this @yuliacech! Verified locally. Left a couple minor comments in the code. Let me know what you think.

yuliacech commented:
Thanks a lot for your review, @alisonelizabeth! Addressed your comments in https://github.com/elastic/kibana/pull/143837/commits/386dd2f4aa8c1d3ef8a5d90ef81f0349360c8554

kellyemurphy commented:
With the **Welcome screen for self managed** can we change that phrase `protect against security threats` to `defend against security threats`?

For the **Home page with guided onboarding link** can we change that phrase `add a sample data set` to `play with a sample data set`? We've already added it, they just need to use it.

alisonelizabeth reviewed (approved):
Thanks for addressing my feedback! Latest LGTM.

yuliacech commented:
Thanks a lot for your review, @kellyemurphy! I updated the texts as you suggested and the screenshots in the PR description.

rshen91 reviewed (approved):
LGTM - tested locally. These changes respect expected behavior of changing the `default route` in advanced settings (it then navigates to that designated page on startup showing the no data page template. I tested changing it to /app/discover# and brought kibana back up without loading any data). Thank you for the comments providing additional context in `add_data.tsx`.

kellyemurphy reviewed (approved):
copy LGTM
